Pay Later button preview is missing (3286)

stracker-phil opened the pull request:
### Problem

In the latest plugin version, there is no button preview displayed in the "Pay Later" tab.

### Cause

The relevant preview script and CSS file were not enqueued in that tab; the deciding condition did not consider the Pay Later tab as for the asset.

### Solution

Updating the condition to enqueue "gateway-settings.js" and its CSS file solves the problem.

stracker-phil committed:
🐛 Load the preview button logic on Pay Later tab

stracker-phil committed:
🔥 Remove unused dependencies

Left overs from PCP-3179

stracker-phil committed:
♻️ Consolidate preview-HTML in a new setting type

stracker-phil committed:
✨ Add preview button eligibility check

stracker-phil committed:
💄 Complete rewrite of the preview box styling

Original styling designed for one use case (variable number of buttons, floating on the far right side).

Meanwhile we have multiple preview boxes, displaying, a fixed or variable amount of buttons; some preview boxes are also positioned inside a “frame” that requires custom padding and responsive checks.

Dinamiko commented:
@stracker-phil PR approved, please fix the CI errors before moving to QA, thanks.

stracker-phil committed:
🚨 Fix reported CI error

stracker-phil committed:
🐛 Hide the PayLater preview when using Vaultig

stracker-phil committed:
🐞 Fix JS error during PayLater preview

Incorrect selector for `payLaterButtonPreview` caused the preview to fail when the “Enable Pay Later Button” was unchecked during page load.

stracker-phil committed:
🥅 Catch several JS/API related preview issues

stracker-phil commented:
I have updated the PR to address issues identified during acceptance tests; primarily JavaScript issues that led to failures in preview generation in certain scenarios.

stracker-phil committed:
🔀 Merge branch 'trunk'
